### PR TITLE
Milestone 2: Add CRUD Functionalities

### PR DESCRIPTION
**Pull request by sntakirutimana72**
## To-Do-List Milestone 2: Add CRUD Functionalities

In this milestone, I

- Moved the app essential logic into a different file and created  the followings:
  - A **populate** function for reading from **localStorage** and populating in the browser.
  - A **create** function for creating a new **To-Do** task and committing it into **localStorage**.
  - A **update** function for committing changes made to the task.
  - A **remove** function for removing task from both **localStorage** and DOM tree.
- Fix linter errors.

### CODE REVIEW

File: src/modules/to-do-manager.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const obj = {
+      index: this.tasks.length,
+      description,
+      completed: false,
+    };
+
+    this.tasks.push(obj);
+    this.listView.appendChild(this.render(obj));
+
+    this.commitTasks();
```
Comment (AnselemOdims) [inline src/modules/to-do-manager.js]: - [x] At the moment, a user is not able to add todo tasks. Kindly ensure that the function to add tasks works as it should.
- [x] Please also ensure that the functionality for editing and deleting works as it should.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @sntakirutimana72 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @sntakirutimana72 , 

Good job so far. I really like what you did!
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] As mentioned by the previous reviewer, kindly ensure that the function to add and remove todos works properly. First make sure you correctly handle local storage

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sntakirutimana72** [comment]
Hello @m05e5 
Kindly provide thorough details of what doesn't work cause everything works at my end.

**Comment by sntakirutimana72** [comment]
Would you please point out what doesn't work at your end so I can get to it?
I don't know what to look for since everything seems to work perfectly at my end.
Kindly acknowledge.

@m05e5

**Comment by rotimiazeez** [comment]
_ @sntakirutimana72 👋🏽, I guess there was a mix up somewhere, but I can say you've completed this milestone 💪🏽.

**Review by rotimiazeez — COMMENTED** [review]
# Hi @sntakirutimana72 👍🏽, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**